File: src/main/java/com/snowofsunflower/android/ui/talkerimpl/AskTalker.java
package com.snowofsunflower.android.ui.talkerimpl;

import android.app.Activity;
import android.view.View;

import com.snowofsunflower.android.ui.talker.IAskTalker;
import com.snowofsunflower.android.ui.talker.Reactor;
import com.yarolegovich.lovelydialog.LovelyStandardDialog;

/**
 * Created by zhouztashin on 2018/10/21.
 */

public class AskTalker implements IAskTalker {

    private LovelyStandardDialog mDialog;

    public AskTalker(Activity activity) {
        mDialog = new LovelyStandardDialog(activity, LovelyStandardDialog.ButtonLayout.HORIZONTAL);
        mDialog.setNegativeButton(android.R.string.cancel, null);
    }

    @Override
    public IAskTalker question(String str) {
        mDialog.setMessage(str);
        return this;
    }

    @Override
    public IAskTalker yes(final Reactor reactor) {
        mDialog.setPositiveButton(android.R.string.ok, new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                reactor.react(view);
            }
        });

        return this;
    }

    @Override
    public IAskTalker no(final Reactor reactor) {
        mDialog.setNegativeButton(android.R.string.cancel, new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                reactor.react(view);
            }
        });

        return this;
    }

    @Override
    public IAskTalker yesText(String text) {
        mDialog.setPositiveButtonText(text);
        return this;
    }

    @Override
    public IAskTalker noText(String text) {
        mDialog.setNegativeButtonText(text);
        return this;
    }

    @Override
    public IAskTalker about(String title) {
        mDialog.setTitle(title);
        return this;
    }

    @Override
    public IAskTalker on() {
        mDialog.show();
        return this;
    }

    @Override
    public IAskTalker off() {
        mDialog.dismiss();
        return this;
    }
}

File: src/main/java/com/snowofsunflower/android/ui/talkerimpl/AskTalker.kt
package com.snowofsunflower.android.ui.talkerimpl

import android.app.Activity
import android.view.View
import com.enjoy.dialog.R
import com.snowofsunflower.android.ui.talker.IAskTalker
import com.yarolegovich.lovelydialog.LovelyStandardDialog

/**
 * 提问对话框实现类
 */
class AskTalker(private val mActivity: Activity) : IAskTalker {
    override fun about(str: String): IAskTalker {
        mDialog.setTitle(str)
        return this
    }

    override fun isOn(): Boolean = mDialog.create().isShowing
    //Done 更改为相应的函数类型？
    var mYesClickListener: ((View, IAskTalker) -> Unit)? = null
    var mNoClickListener: ((View, IAskTalker) -> Unit)? = null
    var mYesText: String? = null
    var mNoText: String? = null

    private val mDialog = LovelyStandardDialog(
            mActivity,
            LovelyStandardDialog.ButtonLayout.HORIZONTAL
    )

    init {
        mDialog.setNegativeButton(R.string.cancel, null)
    }

    override fun question(str: String): IAskTalker {
        mDialog.setMessage(str)
        return this
    }

    override fun yes(reactor: (v: View, talker: IAskTalker) -> Unit): IAskTalker {
        mYesClickListener = reactor
        return this
    }

    override fun no(reactor: (v: View, talker: IAskTalker) -> Unit): IAskTalker {
        mNoClickListener = reactor
        return this
    }

    override fun yesText(text: String): IAskTalker {
        mYesText = text
        //  mDialog.setPositiveButtonText(text)
        return this
    }

    override fun noText(text: String): IAskTalker {
        mNoText = text
        // mDialog.setNegativeButtonText(text)
        return this
    }

    /*  override fun  about(str: String): AskTalker {
          mDialog.setTitle(str)
          return this
      }*/
    /*override fun about(str: String): ITalker {
        mDialog.setTitle(str)
        return this
    }
*/
    override fun on(): IAskTalker {
        val yesText = mYesText ?: mActivity.getString(R.string.ok)
        mYesClickListener?.run {
            mDialog.setPositiveButton(yesText) {
                this(it, this@AskTalker)
            }
        } ?: mDialog.setPositiveButton(yesText, null)

        val noText = mNoText ?: mActivity.getString(R.string.cancel)
        mNoClickListener?.run {
            mDialog.setNegativeButton(mNoText) {
                this(it, this@AskTalker)
            }
        } ?: mDialog.setNegativeButton(noText, null)

        mDialog.show()
        return this
    }

    override fun off(): IAskTalker {
        mDialog.dismiss()
        return this
    }
}
